vsphere/datadog_checks/vsphere/api.py

```diff
@@ -383,3 +392,65 @@ def get_max_query_metrics(self):
             return max_historical_metrics
         else:
             return UNLIMITED_HIST_METRICS_PER_QUERY
+
+    @smart_retry
+    def get_vsan_events(self, timestamp):
+        event_manager = self._conn.content.eventManager
+        entity_time = vim.event.EventFilterSpec.ByTime(beginTime=timestamp)
+        query_filter = vim.event.EventFilterSpec(eventTypeId=VSAN_EVENT_IDS, time=entity_time)
+        events = event_manager.QueryEvents(query_filter)
+        self.log.debug("Received %s vSAN events", len(events))
+        return events
+
+    @smart_retry
+    def get_vsan_metrics(self, cluster_nested_elts, entity_ref_ids, id_to_tags, starting_time):
+        self.log.debug("Querying vSAN metrics: %s", self._vsan_stub)
+        vsan_perf_manager = vim.cluster.VsanPerformanceManager('vsan-performance-manager', self._vsan_stub)
+        health_metrics = []
+        performance_metrics = []
+        for cluster_reference, nested_ids in cluster_nested_elts.items():
+            self.log.debug("Querying vSAN metrics for cluster %s", cluster_reference.name)
+            unprocessed_health_metrics = vsan_perf_manager.QueryClusterHealth(cluster_reference)
+            processed_health_metrics = {}
+            group_id = unprocessed_health_metrics[0].groupId
+            group_health = unprocessed_health_metrics[0].groupHealth
+            processed_health_metrics.update(
+                {
+                    'vsphere.vsan.cluster.health.count': {
+                        'id': group_id,
+                        'status': group_health,
+                        'vsphere_cluster': cluster_reference.name,
+                    }
+                }
+            )
+            for health_test in unprocessed_health_metrics[0].groupTests:
+                test_name = health_test.testId.split('.')[-1]
```

Comment:
Can we validate that `vsan.cluster.health.test_name` is an expected metric name here?

Reply:
What do you mean? Wouldn't checking if the metric name (`'vsphere.vsan.cluster.health.{}.count'.format(test_name)`) is in `VSAN_CLUSTER_METRICS` as I do in the collect_vsan_metrics method of vsphere.py be sufficient validation?

Reply:
Ah yes my bad 👍